proof/crefine/AARCH64/Machine_C.thy

```diff
@@ -19,6 +19,17 @@ instance virq_C :: array_inner_packed
   apply intro_classes
   by (simp add: size_of_def)
 
+context kernel
+begin
+
+(* for doSMC_mop_ccorres *)
+definition smc_args_to_C :: "machine_word eight_tuple \<Rightarrow> smc_args_t__C" where
+  "smc_args_to_C \<equiv>
+     \<lambda>(x0, x1, x2, x3, x4, x5, x6, x7).
+       smc_args_t__C (ARRAY i. [x0, x1, x2, x3, x4, x5, x6, x7] ! i)"
+
+end
```

Comment:
why `context kernel` for this one, instead of dumping it into kernel_m below? I don't remember how long it takes to enter/leave kernel context, but I don't think it's fast; might be a little faster to say `definition (in kernel)` if you really want it to be in kernel here

Reply:
I think I had to do that, because it is also used in theories that are not using `kernel_m`, but just `kernel` (I was surprised to find out these existed, but there is a whole bunch of theories that don't need any machine interface assumptions, and they are not using `kernel_m`).